posthog/models/personal_api_key.py

```diff
@@ -1,15 +1,26 @@
+from django.contrib.auth.hashers import get_hasher
 from django.db import models
 from django.utils import timezone
 
-from .utils import generate_random_token, generate_random_token_personal
+from .utils import generate_random_token
+
+PERSONAL_API_KEY_SALT = "posthog_personal_api_key"
```

Comment:
since we can't change the salt should we provide it from the environment

Might be being over-protective but since it is one salt for all values if someone exfiltrates the database they can crack the keys using a dictionary attack because they know the salt. Whereas if the salt was private then they'd be stuck with brute force...

Reply:
That is true… I thought about using `SECRET_KEY` for this, but the issue with that one is that `SECRET_KEY` is generally not used for persistent password-like things, but rather session tokens or password reset tokens – which are easy to invalidate if the `SECRET_KEY` leaks out. We might not want personal API key to be invalidated at the same time…
Though, on the other hand, if our `SECRET_KEY` does leak out, it will mean something went seriously wrong. It might never happen.
So, for now I updated this to use `SECRET_KEY`, though not 100% sure. @guidoiaquinti What's your opinion on this?

Reply:
Ah, it definitely shouldn't be secret key since the hasher includes the salt as plain text in the generated value 🤣 

![Screenshot 2022-09-22 at 09 02 50](https://user-images.githubusercontent.com/984817/191691998-b2d8345a-3a0d-4df1-89d9-5dfb11df02f1.png)

Also means obscuring it in code doesn't protect against dictionary attacks so 🤷

Reply:
Ah, of course. Well, a salt is only fully effective when it's unique for every record, but that's not feasible for personal API keys. Reverted `SECRET_KEY` change.